packages/backend/src/modules/tracked-txs/TrackedTxsIndexer.ts

```diff
@@ -0,0 +1,220 @@
+  findConfigurationsToSync,
+} from './utils'
+import { diffTrackedTxConfigurations } from './utils/diffTrackedTxConfigurations'
+import { getSafeHeight } from './utils/getSafeHeight'
+
+export type TrackedTxsIndexerUpdaters = Record<
+  TrackedTxsConfigType,
+  TxUpdaterInterface | undefined
+>
+
+export class TrackedTxsIndexer extends ChildIndexer {
+  readonly indexerId = 'tracked_txs_indexer'
+  readonly updaters: Partial<TrackedTxsIndexerUpdaters>
+
+  constructor(
+    logger: Logger,
+    parentIndexer: HourlyIndexer,
+    updaters: TrackedTxsIndexerUpdaters,
+    private readonly trackedTxsClient: TrackedTxsClient,
+    private readonly stateRepository: IndexerStateRepository,
+    private readonly configRepository: TrackedTxsConfigsRepository,
+    private readonly configs: TrackedTxConfigEntry[],
+    private readonly minTimestamp: UnixTime,
+  ) {
+    super(logger, [parentIndexer])
+    this.updaters = pickBy(updaters, notUndefined)
+  }
+
+  override async start(): Promise<void> {
+    await this.initialize()
+    await super.start()
+    this.logger.info('Started')
+  }
+
+  override async update(from: number, to: number): Promise<number> {
+    const { from: unixFrom, to: unixTo } = adjustRangeForBigQueryCall(from, to)
+
+    const [configurations, syncTo] = await this.getConfigurationToSync(
+      unixFrom,
+      unixTo,
+    )
+
+    if (configurations.length === 0) {
+      this.logger.info('Update skipped', { from: unixFrom, to: syncTo })
+      return syncTo.toNumber()
+    }
+
+    const txs = await this.trackedTxsClient.getData(
+      configurations,
+      unixFrom,
+      syncTo,
+    )
+
+    await this.configRepository.runInTransaction(async (trx) => {
+      for (const [type, updater] of Object.entries(this.updaters)) {
+        const filteredTxs = txs.filter((tx) => tx.use.type === type)
+        await updater?.update(filteredTxs, trx)
+      }
+
+      const configIds = configurations.flatMap((config) =>
+        config.uses.map((use) => use.id),
+      )
+      await this.configRepository.setLastSyncedTimestamp(configIds, syncTo, trx)
+    })
+
+    this.logger.info('Updated', {
+      from: unixFrom,
+      to: syncTo,
+      usedConfigurations: configurations.length,
+      fetchedTxsCount: txs.length,
+    })
+    return syncTo.toNumber()
+  }
+
+  async getConfigurationToSync(
+    from: UnixTime,
+    to: UnixTime,
+  ): Promise<[TrackedTxConfigEntry[], UnixTime]> {
+    const databaseEntries = await this.configRepository.getAll()
+    const updaterTypes = Object.keys(this.updaters)
+
+    const { configurationsToSync, syncTo } = findConfigurationsToSync(
+      updaterTypes,
+      this.configs,
+      databaseEntries,
+      from,
+      to,
+    )
+
+    return [configurationsToSync, syncTo]
+  }
+
+  private async initialize(): Promise<void> {
+    const databaseEntries = await this.configRepository.getAll()
+    const { toAdd, toRemove, toTrim } = diffTrackedTxConfigurations(
```

Comment:
we also need to add this "enabled updaters" context to this diffTrackedTxConfigurations functions, we do not want to add a configuration if updater for it is not enabled

Reply:
Why not? We should add it but do not sync.